File: examples/hibernate-jpa-start/src/test/java/be/kuleuven/studenthibernate/domain/StudentPersistenceTest.java
package be.kuleuven.studenthibernate.domain;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import javax.persistence.EntityManager;
import javax.persistence.EntityManagerFactory;
import javax.persistence.Persistence;

public class StudentPersistenceTest {

    private EntityManagerFactory factory;
    private EntityManager entityManager;

    @Before
    public void setUp() {
        this.factory = Persistence.createEntityManagerFactory("be.kuleuven.studenthibernate.domain");
        this.entityManager = factory.createEntityManager();
    }

    @After
    public void tearDown() {
        factory.close();
    }

    @Test
    public void studentCanBePersisted() {
        var jos = new Student("Lowiemans", "Jos", true);
        entityManager.persist(jos);
    }

}

File: examples/kotlin/hibernate-jpa-start/bin/test/be/kuleuven/studenthibernate/domain/StudentPersistenceTest.kt
package be.kuleuven.studenthibernate.domain

import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import javax.persistence.EntityManager
import javax.persistence.EntityManagerFactory
import javax.persistence.Persistence

class StudentPersistenceTest {

    private lateinit var factory: EntityManagerFactory
    private lateinit var entityManager: EntityManager

    @BeforeEach
    fun setUp() {
        factory = Persistence.createEntityManagerFactory("be.kuleuven.studenthibernate.domain")
        entityManager = factory.createEntityManager()
    }

    @AfterEach
    fun tearDown() {
        factory.close()
    }

    @Test
    fun studentCanBePersisted() {
        val jos = Student("Lowiemans", "Jos", true)
        entityManager.persist(jos)
    }
}